runtime/runtime.go

```diff
@@ -149,6 +155,10 @@ type Params struct {
 	// GracefulShutdownPeriod is the time (in seconds) to wait for the http
 	// server to shutdown gracefully.
 	GracefulShutdownPeriod int
+
+	// Incognito flag controls whether OPA will report its version to an external service.
+	// If this flag is true, OPA will not report its version to the external service
+	Incognito bool
```

Comment:
note: since it's on-by-default we'll want to make sure that any e2e tests disable it...I'm not sure there's a good way around this other than being diligent in the e2e tests and maybe using a runtime fixture as much as possible. @patrick-east do you have any thoughts?

Reply:
Yep. The alternative is to keep the old behavior by flipping the meaning of the flag on Params. Have like `EnableVersionCheck` and force callers (cmd package, etc) to flip it to true when they want to enable it.

Reply:
@patrick-east @ashutosh-narkar WDYT? I feel like this would be a good idea--if people are instantiating the Runtime directly they probably don't expect this feature to be enabled.

Reply:
I think having the behavior opt-in at the runtime API level makes sense.

Reply:
Updated runtime to have a `EnableVersionCheck` field. By default, it's `false`. Executing the `opa run` command will still report the version by default.